sources/pmwebapi_test.go

```diff
@@ -55,49 +55,41 @@ func TestUnmarshal(t *testing.T) {
 	}
 }
 
-func TestLabelTypes(t *testing.T) {
-	units := ""
-	pcpType := ""
-	name := "kal"
-	response := typeLabel(units, pcpType, name)
-	expected := "kal"
-	if response != expected {
-		t.Fatalf("Failed when units and type strings were empty. Expected: %s, Got: %s", expected, response)
-	}
-
-	units = ""
-	pcpType = ""
-	name = ""
-	response = typeLabel(units, pcpType, name)
-	expected = ""
-	if response != expected {
-		t.Fatalf("Failed when all strings were empty. Expected: %s, Got: %s", expected, response)
-	}
+func TestWeirdMetrics(t *testing.T) {
+	name := make(map[string]string)
 
-	units = "kal"
-	pcpType = "CoUnTeR"
-	name = "kal"
-	response = typeLabel(units, pcpType, name)
-	expected = "kal_kal_total"
-	if response != expected {
-		t.Fatalf("Failed when type was case sensitive. Expected: %s, Got: %s", expected, response)
+	name["count / sec"] = "count_per_second"
+	name["Kbyte"] = "kilobytes"
+	name["nanosec"] = "nanoseconds"
+	name["millisec"] = "milliseconds"
+	name["failcnt"] = "failcount"
+	name["byte"] = "bytes"
+	name["Mbyte"] = "megabytes"
+	name["sec"] = "seconds"
+	name["bytes_byte"] = "bytes"
+	name["kilobytes"] = "kilobytes"
+	name["seconds"] = "seconds"
+	name["nanoseconds"] = "nanoseconds"
+	name["milliseconds"] = "milliseconds"
+	name["megabytes"] = "megabytes"
+	name["bytes"] = "bytes"
+	for k := range name {
```

Comment:
This looks much better as a map than the previous array. But, to make this even more go-centric, when you iterate over a map with `range`, you iterate over the `key` first, then the `value` second. In other words, if you need both the key and value (which in your case you do), then it is best to setup your for loop as (don't use the variable names `key` and `value` like I do here, use something more appropriate for your setup):
```golang
for key, value := range m {
    ...
}
```
This way, you can replace `name[k]` with `value` (again, using a different naming scheme) for direct access.

Reply:
for this would i defrine value as value = name[k] and use that?

Reply:
No need to define anything, the `for key, value := range m` example I gave you automatically gives you the key (analogous to your `k`) and the value (analogous to your `name[k]`) for every item of your map. These two are always paired as well, so you don't have to worry about getting a value that doesn't match the key.